### PR TITLE
Pc last online

### PR DESCRIPTION
**Pull request by pconrad**
This PR is ultimately from https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-2/pull/35/commits by user 
@chriscerie

This updates the last logged in time immediately once the user logs in and once every minute thereafter. Users logged in within the past 2 minutes are shown as `Online now`. This shows relative time for anything under a week. Anything older will be shown as month/date/year.

This works even when users log in from different time zones. The database stores everyone's time in UTC that later gets translated to each user's local time zone when displaying UI.

There's no storybook available for the admin users page. That will need to be added in a different issue.

Before
image

After
image
image
image
image
image

### CODE REVIEW

File: frontend/src/main/components/Users/UsersTable.js [hunk 1]
```diff
@@ -18,6 +19,10 @@ const columns = [
         Header: 'Email',
         accessor: 'email',
     },
+    {
+        Header: 'Last Online',
+        accessor: 'lastOnline',
+    },
```
Comment (chriscerie) [inline frontend/src/main/components/Users/UsersTable.js:25]:
```suggestion
    {
        Header: 'Last Online',
        id: 'lastOnline',
        accessor: (row) => formatTime(row.lastOnline),
    },
```

File: frontend/src/main/components/Users/UsersTable.js [hunk 2]
```diff
@@ -27,7 +32,12 @@ const columns = [
 
 export default function UsersTable({ users }) {
     return <OurTable
-        data={users}
+        data={users && users.map(user => {
+            return {
+                ...user, 
+                lastOnline: formatTime(user.lastOnline),
+            };
+        })}
```
Comment (chriscerie) [inline frontend/src/main/components/Users/UsersTable.js:40]:
```suggestion
        data={users}
```

**Review by chriscerie — CHANGES_REQUESTED** [review]
Looks like this [diff](https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-2/compare/75f9fb0b1ac2a884929d256c08e432b10dd2eb98..1ddc0df58b9c2df310ecee0468431534038764c1?expand=1) from the pr wasn't merged over. Otherwise unrelated rerenders will break table data shallow equality optimizations

**Comment by pconrad** [comment]
> Looks like this [diff](https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-2/compare/75f9fb0b1ac2a884929d256c08e432b10dd2eb98..1ddc0df58b9c2df310ecee0468431534038764c1?expand=1) from the pr wasn't merged over. Otherwise unrelated rerenders will break table data shallow equality optimizations

Thanks @chriscerie !

File: frontend/src/tests/pages/HomePage.test.js [hunk 3]
```diff
@@ -69,8 +69,6 @@ describe("HomePage tests", () => {
         expect(title).toBeInTheDocument();
         expect(typeof (title.textContent)).toBe('string');
         expect(title.textContent).toEqual('Howdy Farmer');
-
-        expect(() => screen.getAllByTestId(/commonsCard-button/)).toThrow('Unable to find an element');
```
Comment (chriscerie) [inline frontend/src/tests/pages/HomePage.test.js:73]: Is there a reason this assertion was deleted? This feature doesn't the homepage.

Comment (pconrad) [reply frontend/src/tests/pages/HomePage.test.js:73]: Probably an accident while resolving merge conflicts.  I'll see what happens when we restore it.

**Pull request merged**